### PR TITLE
Tests and cleanup for model ensembles

### PR DESCRIPTION
**Pull request by marta-sd**

### CODE REVIEW

File: oddt/scoring/__init__.py [hunk 1]
```diff
@@ -63,6 +80,81 @@ def __init__(self, model_instance, descriptor_generator_instance, score_title='s
...
+                          pdbbind_versions=(2007, 2012, 2013, 2014, 2015, 2016),
+                          desc_path=None):
+        pdbbind_versions = sorted(pdbbind_versions)
+
+        # generate metadata
+        df = None
+        for pdbbind_version in pdbbind_versions:
+            p = pdbbind('%s/v%i/' % (pdbbind_dir, pdbbind_version),
+                        version=pdbbind_version,
+                        opt={'b': None})
```
Comment (mwojcikowski) [inline oddt/scoring/__init__.py]: Switching off bonding is OK for RF-Score, while NNScore needs bonding.

File: oddt/scoring/__init__.py [hunk 2]
```diff
@@ -63,6 +80,81 @@ def __init__(self, model_instance, descriptor_generator_instance, score_title='s
         self.descriptor_generator = descriptor_generator_instance
         self.score_title = score_title
 
+    def _gen_pdbbind_desc(self,
```
Comment (mwojcikowski) [inline oddt/scoring/__init__.py]: Right now we train all of our scoring functions on PDBbind, but I think we should separate the main `scorer` class from `pdbbind_training` scorer class (name subjected to change).

File: tests/test_scoring.py [hunk 3]
```diff
@@ -182,3 +236,34 @@ def test_nnscore():
...
+    rec = next(oddt.toolkit.readfile('pdb', receptor_pdb))
+    rec.protein = True
+    rec.addh()
+
+    data_dir = os.path.join(test_data_dir, 'data')
+    home_dir = mkdtemp()
+    pdbbind_versions = (2007, 2013, 2016)
+
+    for pdbbind_v in pdbbind_versions:
+        os.symlink(os.path.join(data_dir, 'pdbbind'),
```
Comment (mwojcikowski) [inline tests/test_scoring.py]: Creating symlinks will fail if they exist. I'd check for existance here and remove it afterwards if the file is symlink.

**Comment by marta-sd** [comment]
I think I figured out why tests for datasets pass and for scoring functions not. Molecules are loaded in a lazy manner, so the pocket for 1ps3 isn't actually created until we try to use it. So here:
https://github.com/oddt/oddt/blob/62e6898446d347137a02d745d14749afe1d242a6/oddt/toolkits/rdk.py#L362-L367
and here:
https://github.com/oddt/oddt/blob/62e6898446d347137a02d745d14749afe1d242a6/oddt/toolkits/rdk.py#L390-L392
we fail to catch it because `source` is not None, and we end up creating a `Molecule` object from the structure that RDKit can't read

**Pull request merged**